pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,161 @@
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	errSecret := shared.CreateSecret("secret1", "default", false)
+	Expect(errSecret).NotTo(HaveOccurred(), "error creating secret")
+
+	shared.LogLevel("INFO", "TEST: 'CLASSIC' Secrets Encryption method")
+
+	index := len(nodes) - 1
+	cpIp := nodes[index].ExternalIP
+	secretsEncryptOps("prepare", product, cpIp, nodes)
+	secretsEncryptOps("rotate", product, cpIp, nodes)
+	secretsEncryptOps("reencrypt", product, cpIp, nodes)
+
+	if strings.Contains(os.Getenv("TEST_TYPE"), "both") {
+		shared.LogLevel("INFO", "TEST: 'NEW' Secrets Encryption method")
+		secretsEncryptOps("rotate-keys", product, cpIp, nodes)
+	}
+}
+
+func secretsEncryptOps(action, product, cpIp string, nodes []shared.Node) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyActionStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
+		time.Sleep(20 * time.Second) // Wait for reencrypt action to complete before restarting services
+	}
+	for i, node := range nodes {
+		nodearr := []string{node.ExternalIP}
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
+		Expect(errRestart).NotTo(HaveOccurred(), "error restart service for node: "+nodeIp)
+		// Order of reboot matters. Etcd first then control plane nodes.
+		// Little lag needed between node restarts to avoid issues.
+		waitEtcdErr := shared.WaitForPodsRunning(5, 4)
+		if waitEtcdErr != nil {
+			shared.LogLevel("WARN", "pods not up after 20 seconds.")
+			if i != len(nodes)-1 {
+				shared.LogLevel("DEBUG", "continue service restarts")
+			}
+		}
+	}
+	switch product {
+	case "k3s":
+		waitPodsErr := shared.WaitForPodsRunning(5, 6)
+		if waitPodsErr != nil {
+			shared.LogLevel("WARN", "pods not up after 30 seconds")
+		}
+	case "rke2":
+		waitPodsErr := shared.WaitForPodsRunning(5, 12)
+		if waitPodsErr != nil {
+			shared.LogLevel("WARN", "pods not up after 60 seconds")
+		}
+	}
+
+	secretEncryptStatus, errGetStatus := waitForHashMatch(cpIp, product, 5, 36) // Max 3 minute wait time for hash to match
+	Expect(errGetStatus).NotTo(HaveOccurred(), "error getting secret-encryption status")
+	verifyStatusStdOut(action, secretEncryptStatus)
+
+	errLog := logEncryptionFileContents(nodes, product)
+	Expect(errLog).NotTo(HaveOccurred())
+}
+
+func waitForHashMatch(cpIp, product string, defaultTime time.Duration, times int) (string, error) {
```

Comment:
if this is a private function, and you are only using in one context, why pass default time as parameter instead of just use the value u want on the sleep ?

Reply:
we should never hard code. that is not good practice. even if it is only used in one place. 
also it follows with the wait pod method pattern. so it is not confusing to people.

Reply:
sorry this is not hard coding....actually thinking on your perspctive what u doing is hard coding as well, since the only difference is that instead of making it simple and where it should be used its passed already, you are passing a value that never changes as parameter..... so in your POV this is also  hard coding, but instead of declaring it on the usage you are declaring elsewhere, and passing it along.

but regardless, this is not hard coding at all, this is you using a value that has no interaction anywhere else, like no one in any way can change this value right? so you are just coding something and using a spefic value for it, which is totally normal. Now if you have a value that can/should/probably needs to be changed,overwrited or whatever , and you are adding the value fixed there, THEN its hard coding.

Reply:
yeah there's not a right or wrong answer here, more of a "which paradigm do we want to use"? In the case of `waitForX` functions, we have one already merged in the framework today `WaitForInstanceRunning` which sets the time vars in the function. This PR has two new ones: `waitForHashMatch` and `WaitForPodsRunning` which both set the time vars as part of the function signature. 

Having it as part of the function signature allows other areas to call with different wait times. Having it as defined within the function says "wait N time for X". If we expect to use these functions a bit more, I personally like having it as part of the function signature, because it's likely that not everything will need to wait the same amount of time. But if we don't expect to use it anymore, or if we expect that future uses require the same amount of wait time, then we should definitely define the time within the function itself.

Reply:
no the point here its that this func its not public, so it will only be consumed in this specific case, and on this specific case its been used a single value with no way to override or change...
so no need to pass as parameter, it only complicates more the scenario

Reply:
That's true yeah, this function itself is private, indicating it will only be used here, so I agree the time parameters should be defined within the function!

Reply:
fixed